### PR TITLE
Mutation/ticket check in

### PR DESCRIPTION
**Pull request by walkerkit**
Hi James,
I think this is all sorted, I've implemented the automated testing of the error cases!

Please let me know if I've missed anything 

:)

### CODE REVIEW

File: uobtheatre/bookings/models.py
Comment (JElgar) [inline uobtheatre/bookings/models.py:458]: Why do you return `self` here?

File: uobtheatre/bookings/schema.py
Comment (JElgar) [inline uobtheatre/bookings/schema.py:73]: I'm not sure why this is back, I think I changed this in a previous pr but I might be wrong

File: uobtheatre/bookings/schema.py
Comment (JElgar) [inline uobtheatre/bookings/schema.py:365]: I would say its slightly easier to read when you don't use `else` and just lead with the exception. I.e.

```
        if booking.performance != performance:
            raise GQLFieldException
        do_other_things
```

File: uobtheatre/bookings/test/factories.py
Comment (JElgar) [inline uobtheatre/bookings/test/factories.py:64]: What happens if we just remove this line?

File: uobtheatre/bookings/test/test_models.py
Comment (JElgar) [inline uobtheatre/bookings/test/test_models.py:923]: Do we need these two factories? Or can we just do
```suggestion
    ticket = TicketFactory(
        checked_in=initialState
    )
```

File: uobtheatre/bookings/test/test_models.py
Comment (JElgar) [inline uobtheatre/bookings/test/test_models.py:912]: :heart:

File: uobtheatre/bookings/test/test_schema.py
Comment (JElgar) [inline uobtheatre/bookings/test/test_schema.py:1126]: We no longer need to do this assertion
```suggestion
```

File: uobtheatre/bookings/test/test_schema.py
Comment (JElgar) [inline uobtheatre/bookings/test/test_schema.py:1129]:
```suggestion
   """
    What are we testing?
    that only the expected tickets are checked in
    that incorrect ticket refs are not checked in
    """
```

File: uobtheatre/utils/test/test_utils_utils.py
Comment (JElgar) [inline uobtheatre/utils/test/test_utils_utils.py:13]: I feel like this also arrived on a different branch :thinking:

**Review by JElgar — CHANGES_REQUESTED** [review]
Very nice, looks really good. 

There seem to be a small git thing going on where some changes which I dont think should be in this pr are. I dont think thats your fault but we should take a look before merging.

File: uobtheatre/bookings/models.py
Comment (walkerkit) [reply uobtheatre/bookings/models.py:458]: I'm guessing this is copied, will nuke it!

File: uobtheatre/bookings/schema.py
Comment (walkerkit) [reply uobtheatre/bookings/schema.py:73]: Wild, I think I branched from main... resolve in the merge?

File: uobtheatre/bookings/schema.py
Comment (walkerkit) [reply uobtheatre/bookings/schema.py:365]: Right, because the exception will break out from the process flow! Will change.

File: uobtheatre/bookings/test/factories.py
Comment (walkerkit) [reply uobtheatre/bookings/test/factories.py:64]: Guessing that's been left in error!

File: uobtheatre/utils/test/test_utils_utils.py
Comment (walkerkit) [reply uobtheatre/utils/test/test_utils_utils.py:13]: This was part of the problem solving, will remove!

Comment (JElgar) [reply uobtheatre/utils/test/test_utils_utils.py:13]: Oh sorry I forgot this was in this branch! No harm in keeping it tbh

Comment (streth11) [reply uobtheatre/utils/test/test_utils_utils.py:13]: is this test necessary?
I can see why it's there and it's harmless to have it - but surely we are assuming the module's testing is rigorous and even if it isn't this single assertion wouldn't have enough rigour to be valid anyway.

If it's just for coverage tho then i'm totally for it xox

Comment (JElgar) [reply uobtheatre/utils/test/test_utils_utils.py:13]: Its not needed but its a nice to have. Until this pr, if someone had just changed `create_short_uuid` to return `1` all tests would have passed. This at least checks a very low level of uniqueness.

Comment (streth11) [reply uobtheatre/utils/test/test_utils_utils.py:13]: fair xx

File: uobtheatre/bookings/test/factories.py
Comment (walkerkit) [reply uobtheatre/bookings/test/factories.py:64]: the world keeps on spinning

**Comment by walkerkit** [comment]
I think this is what you are looking for
Just need to do the merge if you're happy!

File: uobtheatre/bookings/test/factories.py
Comment (JElgar) [reply uobtheatre/bookings/test/factories.py]: :partying_face:

File: uobtheatre/bookings/schema.py
Comment (JElgar) [inline uobtheatre/bookings/schema.py:364]: To be consistent we should probably do the same style here:
```suggestion
            if ticket.booking != booking:
                raise GQLFieldException(
                    message="The ticket booking does not match the mutation booking."
                )
                
            ticket.check_in()
```

**Review by JElgar — CHANGES_REQUESTED** [review]
One tiny thing which might actually just be worse than how you've done it?
Other than that looks great thank you! Let do the merge together sometime tomorrow/next week?

**Review by JElgar — APPROVED** [review]

**Pull request merged**
